firehose/parsers/flawfinder.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python
+#
+#   Copyright 2017 David Carlos  <ddavidcarlos1392@gmail.com>
+#   This library is free software; you can redistribute it and/or
+#   modify it under the terms of the GNU Lesser General Public
+#   License as published by the Free Software Foundation; either
+#   version 2.1 of the License, or (at your option) any later version.
+#
+#   This library is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#   Lesser General Public License for more details.
+#
+#   You should have received a copy of the GNU Lesser General Public
+#   License along with this library; if not, write to the Free Software
+#   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301
+#   USA
+
+import sys
+import re
+from subprocess import check_output
+from firehose.model import Message, Point, \
+    File, Location, Generator, Metadata, Analysis, Issue
+
+
+def main():
+    """ Main entry to flawfinder parser """
+    try:
+        arg_file = sys.argv[1]
+        report = open(arg_file, 'r')
+    except IOError as e:
+        raise e
+    except IndexError:
+        print("Missing input file")
+    analysis = parse_file(report)
+    sys.stdout.write(str(analysis.to_xml()))
+    sys.stdout.write('\n')
+
+
+def parse_file(infile):
+    """ Parser flawfinder output
+
+    :infile: file-like object
+    :returns: Firehose Analysis object, representing the final XML.
+
+    Flawfinder can generate multiple cwes for a single issue.
+    Firehose's models does not supports multiple CWEs.
+    For now, when  multiple CWEs ocurrs, we get only the first one.
+
+    A issue was created to track this bug:
+    https://github.com/fedora-static-analysis/firehose/issues/35
+    """
+
+    line = infile.readline()
+    generator = Generator(name='flawfinder',
+                          version=get_flawfinder_version(line))
+    metadata = Metadata(generator, None, None, None)
+    analysis = Analysis(metadata, [])
+
+    pattern = "\([a-z]*\)"
+    pattern2 = "\[[0-9]\]"
+    prog = re.compile(pattern)
+    prog2 = re.compile(pattern2)
+
+
+    while line:
+        if line != "\n" and (prog.search(line) and prog2.search(line)):
+
+            testid = re.search("\(([a-z]+)\)", line).group(1)
+            issue_path = line.split(":")[0]
+            issue_line = line.split(":")[1]
+
```

Comment:
Thanks for doing the earlier cleanups; the code is clearer now.

With that, the handling of ``prog``, ``prog2`` and the ``testid`` looks a bit unusual; presumably these always appear on the same line, in a consistent way.  Given that, I think it would be improved by using a single regular expression; something like:

    # We want to match a line like:
    #    ./docs/examples/asiohiper.cpp:78:  [4] (shell) system:
    filename_group = "(.+)"
    linenum_group = "([0-9]+)
    severity_group = "\[([0-9])\]"
    category_group = "\((.+)\)"
    funcname_group = "(\S+)"
    ws = '\s+'
    PAT= (filename_group + ':' + linenum_group + ':' + ws + severity_group + ws
          + category_group + ws + funcname_group)

It should then be possible to do:

    m = re.match(PAT, message_line)
    if m:
      # do stuff with m.groups()